File: page-objects/java-testng/complete/src/main/java/pages/HomePage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class HomePage {

    private WebDriver webDriver;

    @FindBy(id = "searchContent")
    private WebElement searchField;

    public HomePage(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(this.webDriver, this);
    }

    public void visit() {
        this.webDriver.get("https://www.zalando.de/");
    }

    public SearchResults search(String searchText) {
        searchField.sendKeys(searchText);
        searchField.submit();
        return new SearchResults(this.webDriver);
    }
}

File: page-objects/js-mocha-chai/initial/pages/home-page.js
var webDriver = require('selenium-webdriver');
var By = webDriver.By;

var SearchResultsPage = require('./search-results-page.js');

HomePage = function HomePage(driver) {
    this.driver = driver;
    this.searchField = By.id("searchContent");
};

HomePage.prototype.visit = function() {
    this.driver.get('https://www.zalando.de/');
    return webDriver.promise.fulfilled(true);
};

HomePage.prototype.search = function(searchText) {
    this.driver.findElement(this.searchField).sendKeys(searchText);
    this.driver.findElement(this.searchField).submit();
    return new SearchResultsPage(this.driver);
};

module.exports = HomePage;